src/vm/vm_core.rs

```diff
@@ -3302,4 +3306,26 @@ mod tests {
             Err(MemoryError::GetRangeMemoryGap)
         );
     }
+
+    #[test]
+    fn get_segment_used_size_after_computing_used() {
+        let mut vm = vm!();
+        vm.memory = memory![
+            ((0, 2), 1),
+            ((0, 5), 1),
+            ((0, 7), 1),
+            ((1, 1), 1),
+            ((2, 2), 1),
+            ((2, 4), 1),
+            ((2, 7), 1)
+        ];
+        vm.segments.compute_effective_sizes(&vm.memory);
+        assert_eq!(Some(8), vm.segments.get_segment_used_size(2));
```

Comment:
```suggestion
        assert_eq!(Some(8), vm.get_segment_used_size(2));
```
The new method is not being tested

Reply:
Fixed